### PR TITLE
fix-tests: expect is not assert

### PR DESCRIPTION
**Pull request by jeffkaufman**
With Chai, instead of `expect(condition)` you need to use `expect(variable).stuff.in.english` or `assert(condition)`.

In cases where the tests were silently failing I've used `iter.skip` for the whole test, or commented out the check if only part of the test was being skipped.  (I don't like commenting out code, but is there a better way to skip just a small part of a test?)

Fixes https://github.com/ampproject/amphtml/issues/13422

### CODE REVIEW

**Comment by rsimha-amp** [comment]
@jeffkaufman there's one lint error due to a line you commented out. https://travis-ci.org/ampproject/amphtml/jobs/339933011#L712

**Comment by rsimha-amp** [comment]
Here are all the `expect` errors that need to be fixed. @jeffkaufman you could either update this PR, or I could do it in a separate PR.

```
~/src/amphtml$ gulp lint
[10:39:58] Using gulpfile ~/src/amphtml/gulpfile.js
[10:39:58] Starting 'lint'...
[10:40:09] Linter error:
/usr/local/google/home/rsimha/src/amphtml/test/functional/test-ads-config.js
  31:7  error  expect(...) used without assertion  chai-expect/missing-assertion

✖ 1 problem (1 error, 0 warnings)


[10:40:10] Linter error:
/usr/local/google/home/rsimha/src/amphtml/test/functional/test-custom-element-registry.js
  284:14  error  operator ">" used in expect(), use "to.be.above()" instead  chai-expect/no-inner-compare

✖ 1 problem (1 error, 0 warnings)


[10:40:11] Linter error:
/usr/local/google/home/rsimha/src/amphtml/test/functional/test-custom-element.js
  182:7  error  expect(...) used without assertion  chai-expect/missing-assertion
  207:7  error  expect(...) used without assertion  chai-expect/missing-assertion

✖ 2 problems (2 errors, 0 warnings)


[10:40:12] Linter error:
/usr/local/google/home/rsimha/src/amphtml/test/functional/test-iframe-helper.js
  147:13  error  expect(...) used without assertion  chai-expect/missing-assertion

✖ 1 problem (1 error, 0 warnings)


[10:40:13] Linter error:
/usr/local/google/home/rsimha/src/amphtml/test/functional/test-json.js
  112:14  error  operator "===" used in expect(), use "to.equal()" instead  chai-expect/no-inner-compare
  120:14  error  operator "===" used in expect(), use "to.equal()" instead  chai-expect/no-inner-compare
  129:14  error  operator "===" used in expect(), use "to.equal()" instead  chai-expect/no-inner-compare
  132:14  error  operator "===" used in expect(), use "to.equal()" instead  chai-expect/no-inner-compare

✖ 4 problems (4 errors, 0 warnings)


[10:40:16] Linter error:
/usr/local/google/home/rsimha/src/amphtml/test/functional/test-xhr.js
  471:18  error  operator "===" used in expect(), use "to.equal()" instead  chai-expect/no-inner-compare
  489:18  error  operator "===" used in expect(), use "to.equal()" instead  chai-expect/no-inner-compare
  507:18  error  operator "===" used in expect(), use "to.equal()" instead  chai-expect/no-inner-compare

✖ 3 problems (3 errors, 0 warnings)


[10:40:28] Linter error:
/usr/local/google/home/rsimha/src/amphtml/ads/google/a4a/test/test-utils.js
  473:7   error  expect(...) used without assertion                              chai-expect/missing-assertion
  588:16  error  operator ">=" used in expect(), use "to.be.at.least()" instead  chai-expect/no-inner-compare
  606:16  error  operator ">=" used in expect(), use "to.be.at.least()" instead  chai-expect/no-inner-compare
  619:16  error  operator ">=" used in expect(), use "to.be.at.least()" instead  chai-expect/no-inner-compare
  631:16  error  operator ">=" used in expect(), use "to.be.at.least()" instead  chai-expect/no-inner-compare

✖ 5 problems (5 errors, 0 warnings)


[10:40:28] Linter error:
/usr/local/google/home/rsimha/src/amphtml/extensions/amp-a4a/0.1/test/test-amp-a4a.js
   193:12  error  operator ">" used in expect(), use "to.be.above()" instead  chai-expect/no-inner-compare
  1413:9   error  expect(...) used without assertion                          chai-expect/missing-assertion
  1430:9   error  expect(...) used without assertion                          chai-expect/missing-assertion

✖ 3 problems (3 errors, 0 warnings)


[10:40:28] Linter error:
/usr/local/google/home/rsimha/src/amphtml/extensions/amp-a4a/0.1/test/test-real-time-config-manager.js
   84:14  error  operator ">" used in expect(), use "to.be.above()" instead     chai-expect/no-inner-compare
   86:14  error  operator "<=" used in expect(), use "to.be.at.most()" instead  chai-expect/no-inner-compare
   87:14  error  operator ">" used in expect(), use "to.be.above()" instead     chai-expect/no-inner-compare
  107:11  error  expect(...) used without assertion                             chai-expect/missing-assertion

✖ 4 problems (4 errors, 0 warnings)


[10:40:29] Linter error:
/usr/local/google/home/rsimha/src/amphtml/extensions/amp-ad-network-adsense-impl/0.1/test/test-amp-ad-network-adsense-impl.js
  431:16  error  operator ">=" used in expect(), use "to.be.at.least()" instead  chai-expect/no-inner-compare

✖ 1 problem (1 error, 0 warnings)


[10:40:30] Linter error:
/usr/local/google/home/rsimha/src/amphtml/extensions/amp-ad-network-adzerk-impl/0.1/test/test-amp-ad-network-adzerk-impl.js
  123:20  error  operator "==" used in expect(), use "to.equal()" instead  chai-expect/no-inner-compare
  127:20  error  operator "==" used in expect(), use "to.equal()" instead  chai-expect/no-inner-compare

✖ 2 problems (2 errors, 0 warnings)


[10:40:30] Linter error:
/usr/local/google/home/rsimha/src/amphtml/extensions/amp-ad-network-doubleclick-impl/0.1/test/test-amp-ad-network-doubleclick-impl.js
  1016:7   error  expect(...) used without assertion  chai-expect/missing-assertion
  1026:7   error  expect(...) used without assertion  chai-expect/missing-assertion
  1035:7   error  expect(...) used without assertion  chai-expect/missing-assertion
  1044:7   error  expect(...) used without assertion  chai-expect/missing-assertion
  1403:13  error  expect(...) used without assertion  chai-expect/missing-assertion
  1412:13  error  expect(...) used without assertion  chai-expect/missing-assertion
  1420:13  error  expect(...) used without assertion  chai-expect/missing-assertion
  1428:13  error  expect(...) used without assertion  chai-expect/missing-assertion

✖ 8 problems (8 errors, 0 warnings)


[10:40:40] Linter error:
/usr/local/google/home/rsimha/src/amphtml/extensions/amp-sidebar/0.1/test/test-amp-sidebar.js
  136:9  error  expect(...) used without assertion  chai-expect/missing-assertion

✖ 1 problem (1 error, 0 warnings)


[10:40:40] Linter error:
/usr/local/google/home/rsimha/src/amphtml/extensions/amp-video/0.1/test/test-amp-video.js
  630:11  error  expect(...) used without assertion  chai-expect/missing-assertion
  631:11  error  expect(...) used without assertion  chai-expect/missing-assertion
  650:11  error  expect(...) used without assertion  chai-expect/missing-assertion

✖ 3 problems (3 errors, 0 warnings)


[10:40:41] Error: 39 linter error(s) found.
```

File: test/functional/test-json.js [hunk 1]
```diff
@@ -126,10 +126,10 @@ describe('json', () => {
       const original = {child: {str: 'A', num: 1, bool: true, val: null}};
       const copy = recreateNonProtoObject(original);
       expect(copy).to.deep.equal(original);
-      expect(copy === original).to.be.false;
+      assert(copy !== original);
```
Comment (rsimha-amp) [inline test/functional/test-json.js]: Shouldn't this be `expect(copy).to.not.equal(original)`? (Here and everywhere else in the PR)

Comment (rsimha-amp) [reply test/functional/test-json.js]: See http://chaijs.com/api/bdd/#method_equal

Comment (jeffkaufman) [reply test/functional/test-json.js]: fixed!

File: extensions/amp-a4a/0.1/test/test-real-time-config-manager.js [hunk 2]
```diff
@@ -104,7 +104,7 @@ describes.realWin('real-time-config-manager', {amp: true}, env => {
         expect(rtcResponseArray.length).to.equal(expectedRtcArray.length);
         expect(fetchJsonStub.callCount).to.equal(calloutCount);
         (expectedCalloutUrls || []).forEach(url => {
-          expect(fetchJsonStub.calledWith(url));
+          expect(fetchJsonStub.calledWith(url)).to.be.true;
```
Comment (rsimha-amp) [inline extensions/amp-a4a/0.1/test/test-real-time-config-manager.js]: This should be:
```
expect(fetchJsonStub).to.have.been.calledWith(url);
```

At that point, you might have to disable the test `'should favor publisher URLs over vendor URLs'` which appeared to be silently failing in the past.

File: test/functional/test-json.js [hunk 3]
```diff
@@ -126,10 +126,10 @@ describe('json', () => {
       const original = {child: {str: 'A', num: 1, bool: true, val: null}};
       const copy = recreateNonProtoObject(original);
       expect(copy).to.deep.equal(original);
-      expect(copy === original).to.be.false;
+      expect(copy).to.not.equal(original);
       expect(copy.__proto__).to.be.undefined;
       expect(copy.child).to.deep.equal(original.child);
-      expect(copy.child === original.child).to.be.false;
+      assert(copy.child).to.not.equal(original.child);
```
Comment (rsimha-amp) [inline test/functional/test-json.js]: This should be an `expect`, not an `assert`:
```
expect(copy.child).to.not.equal(original.child);
```

**Comment by rsimha-amp** [comment]
And if you change the last line of the PR description from `Addresses` to `Closes` or `Fixes`, merging the PR will automatically close the associated issue. See https://help.github.com/articles/closing-issues-using-keywords/

**Comment by rsimha-amp** [comment]
@jeffkaufman This is now merged. Thanks for the fix!

**Pull request merged**